pkg/license/download.go

```diff
@@ -138,54 +139,42 @@ func (ddi *DefaultDownloaderImpl) SetOptions(opts *DownloaderOptions) {
 	ddi.Options = opts
 }
 
+//go:embed licenseData
```

Comment:
We have to run the mage target "downloadLicenseData" , so it gets downloaded

Reply:
This is what we need to do in  the prow job too. We'll need to rub this target before running the tests

Reply:
ok, but how we will distribute that when we do a release?

Reply:
I think for release we would do `downloadLicenseData` before building it. 
I want to keep the target for downloading the licenses and the target for building just the binary separate, for quick development. We don't want to download all licenses just to build the binary.

Reply:
ok, but we want to download all the data when we build the target  `buildstaging` 

also, when we release the tool, how the licenses get updated? we will need to release a new version? is that what we want? 

cc @puerco

Reply:
OK, here's how I think it should work:

- First, we should commit the license list. We don't need to re-download it at every build. But what we want is to keep it up to date.  So..
- I think it is a great idea to check the license version in CI, but instead of rebuilding it every time, we want to check we have the latest version.
- @cpanato at some point we could do a new release if there is a new version of the licenses but I think just catching and updating it in a presubmit on each PR is good enough. If we feel we need more we could add a cronjoib to check but I don't think it changes enough to warrant it.
 
One way of doing it would be to download and rebuild the JSON data file and compare if there is a diff, but I feel we are confident in the license downloader, then we can check the version to make it more efficient.

Reply:
Also, let's make sure we document how to update the license list, which would be something like:

### In CI:

- Contributor opens a normal code PR (not license related)
- CI checks the license list to find out the newest version on spdx and compares it to `  "licenseListVersion": "3.18"`
- If its newer, CI b0rks, throws a helpful error instructing how to update

### Then the Contributor:

Something like this:

- Runs mage downloadLicenseData
- `git add pkg/license/licenseData/`
- `git commit -sm "Update SPDX Lincese List to $(cat  pkg/license/licenseData/licenseList.json | jq -r .licenseListVersion)"`

push and everything should work

Reply:
@puerco should this CI job be a prow job or a github action ?

Reply:
Mmh good question @cpanato WDYT?